app/assets/javascripts/sitewide/drag_n_drop.js.erb

```diff
@@ -0,0 +1,350 @@
+      html += '<span class="glyphicon glyphicon-file"></span>';
+      html += '<%= I18n.t 'assets.drag_n_drop.file_label' %>';
+      html += '<div class="pull-right">';
+      html += '<a data-item-id="' + i + '" href="#">';
+      html += '<span class="glyphicon glyphicon-remove"></span></a></div></div>';
+      html += '<div class="panel-body"><div class="form-group">';
+      html += '<label class="control-label">Name</label>';
+      html += '<input type="text" class="form-control" ';
+      html += 'onChange="DragNDropResults.validateTextSize(this)"';
+      html += ' rel="results[name]" name="results[name][' + i + ']">';
+      html += '</div><div class="form-group"><label class="control-label">';
+      html += '<%= I18n.t 'assets.drag_n_drop.file_label' %>:</label> ';
+      html += truncateLongString(asset.name,
+                                 <%= Constants::FILENAME_TRUNCATION_LENGTH %>);
+      html += _validateFilesSize(asset);
+      html += '</div></div>';
+      return html;
+    }
+
+    function _removeItemHandler(id) {
+      $('[data-item-id="' + id +'"]').off('click').on('click', function(e) {
+        e.preventDefault();
+        e.stopImmediatePropagation();
+        e.stopPropagation();
+        var $el = $(this);
+        var index = $el.data('item-id');
+        totalSize -= parseInt(droppedFiles[index]/1048576);
+        droppedFiles[index] = null;
+        $el.closest('.panel-result-attachment-new').remove();
+        _validateTotalSize();
+      });
+    }
+
+    function uploadResultAssets(button) {
+      if(isValid && _filerAndCheckFiles()) {
+        animateSpinner();
+        $.ajax({
+          url: $(button).attr('data-href'),
+          method: 'POST',
+          data: _appendFilesToForm(),
+          contentType: false,
+          processData: false,
+          success: function(data) {
+            animateSpinner(null, false);
+            $('#new-result-assets-select').parent().remove();
+            $(data.html).prependTo('#results').promise().done(function() {
+              $.each($('[data-container="new-reports"]').find('.result'),
+                     function() {
+                initFormSubmitLinks($(this));
+                ResutlAssets.applyEditResultAssetCallback();
+                Results.applyCollapseLinkCallBack();
+                Results.toggleResultEditButtons(true);
+                initPreviewModal();
+                Comments.initialize();
+                ResutlAssets.initNewResultAsset();
+                Results.expandResult($(this));
+              });
+
+            });
+            $('#results-toolbar').show();
+          },
+          error: function() {
+            animateSpinner();
+            location.reload();
+          }
+        })
+      }
+    }
+
+    function destroyAll() {
+      _dragNdropAssetsOff();
+      droppedFiles = [];
+      isValid = true;
+      totalSize = 0;
+    }
+
+    function _dragNdropAssetsOff() {
```

Comment:
Some functions just like this one are duplicated. Can you do this without duplicates for ALL such similar functions?